### PR TITLE
Cyborgs only die once when dead, and don't break binary talk

### LINKED ISSUE #2616: Binary Communications Loop.
After severe damage in a borg body, sometimes when you're brought back online... the binary communications loops so badly the game crashes.
It'll just repeat the messages like a hundred times.

### LINKED ISSUE #2747: Cyborgs are VERY STRANGE and VERY GLITCHY
Being shot repeatedly as a cyborg got me "Killed" - The health indicator was a bluescreen, and it said I shuddered violently for a moment, before becoming motionless.

However, it repeated that death message twice a second for as long as my battery was still in me, and I could still walk around, despite being able to see through walls.

Not only that, but I could still talk in cyborg chat, except doing so made it repeat my message an exponential number of times. The first time, 5. Then, half the text chat. Then, it filled the whole screen. and more. 

This binary glitch persisted even after being repaired.

Update 1: Side note, the binary glitch may be clientside.

Update 2: Being removed from the cyborg body and placed in a new one fixed the glitch. What the hell.

### PR DESCRIPTION
**Pull request by Crazylemon64**
Sure, it was a quick enough fix
BE CAREFUL WHEN SETTING `stat`!
Fixes #2747 
Fixes #2616

### CODE REVIEW

File: code/modules/mob/living/silicon/robot/robot_damage.dm [hunk 1]
```diff
@@ -4,8 +4,15 @@
 		stat = CONSCIOUS
 		return
 	health = maxHealth - (getBruteLoss() + getFireLoss())
```
Comment (tigercat2000) [inline code/modules/mob/living/silicon/robot/robot_damage.dm]: Whoa, whoa, what? 

Somehow missed this. But uh.
Robot/Life() specifically recalculated it the other way around... why bay why

**Comment by Plotron** [comment]
Godlike debugging.

File: code/modules/mob/living/silicon/robot/robot_damage.dm (hunk 1 above)
Comment (Crazylemon64) [inline code/modules/mob/living/silicon/robot/robot_damage.dm]: I found two, and wasn't sure which of them to keep. I'll keep the other one, then.

Comment (Crazylemon64) [inline code/modules/mob/living/silicon/robot/robot_damage.dm]: Wait, I assume you're talking about life.dm having another updateHealth proc, yes?

Comment (tigercat2000) [inline code/modules/mob/living/silicon/robot/robot_damage.dm]: @Crazylemon64 I made that other updateHealth proc- but before I made it, that health setting was directly in Life(), yet this has always existed. I don't understand why.

File: code/modules/mob/living/silicon/robot/robot_damage.dm [hunk 2]
```diff
@@ -1,11 +1,18 @@
-/mob/living/silicon/robot/updatehealth()
+mob/living/silicon/robot/updatehealth()
```
Comment (tigercat2000) [inline code/modules/mob/living/silicon/robot/robot_damage.dm]: would I sound nitpicky to point out that this should actually have a `/` before it
`/mob/living/silicon/robot/updatehealth()`

Comment (Crazylemon64) [inline code/modules/mob/living/silicon/robot/robot_damage.dm]: Ah, whoops. Will fix

**Comment by Crazylemon64** [comment]
In a later PR, I should probably replace all settings of `stat` with functions that work only in certain states, so you don't get immortal robots, or other weird badness

**Pull request merged**